### PR TITLE
Remove backwards-compat with Aesara/PyMC v4

### PR DESCRIPTION
**Pull request by michaelosthege**
To simplify type checking down the road.

### CODE REVIEW

**Comment by lhelleckes** [comment]
Is there a direct need to remove the backwards compatibility? I don't feel like the current version makes things too hard. Do you have features in mind that require a simpler testing pipeline?

**Comment by michaelosthege** [comment]
I'd like to make `mypy` happy with our type annotations and it's going to be a lot simpler if the tensor types come from just one optional dependency.
Also it saves on precious CI pipeline minutes..

Other than that I don't think there's much value in continuing to support an outdated PyMC version. Someone could always pin to `calibr8==7.0.0` if needed.

**Review by lhelleckes — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -24,13 +24,9 @@ jobs:
...
       run: |
         pytest --cov=./calibr8 --cov-append --cov-report xml --cov-report term-missing calibr8/tests.py
-    - name: Install and test with PyMC v4
-      run: |
-        pip install "pymc==4.4.0"
-        pytest --cov=./calibr8 --cov-append --cov-report xml --cov-report term-missing calibr8/tests.py
     - name: Install and test with PyMC v5
       run: |
-        pip install "pymc==5.0.0"
+        pip install "pymc>=5.0.0"
```
Comment (lhelleckes) [inline .github/workflows/release.yml:29]: This might be a bit dangerous in case PyMC has breaking changes?

File: calibr8/core.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 from . import utils
 from .utils import pm
 
-__version__ = "7.0.0"
+__version__ = "7.1.0"
```
Comment (lhelleckes) [inline calibr8/core.py:22]: Isn't that technically a breaking change?

File: .github/workflows/release.yml (hunk 1 above)
Comment (michaelosthege) [reply .github/workflows/release.yml:29]: Then at least we'll notice..

Ideally we should set up a dependabot config to have it open a PR with every new PyMC release: https://github.com/michaelosthege/mcbackend/blob/main/.github/dependabot.yml

Do you want to try dependabot?

File: calibr8/core.py (hunk 2 above)
Comment (michaelosthege) [reply calibr8/core.py:22]: Yes, but not for the model functions. Realistically, no one is using calibr8 with PyMC v4 and I'd like to NOT follow this up with changes to all cmodel JSONs we have internally 🙄

**Pull request merged**